internal/files/linkedfiles.go

```diff
@@ -0,0 +1,408 @@
+	}
+	l.LinkChecksum = l.IncludedFileContentsChecksum
+	l.UpToDate = true
+	return true, nil
+}
+
+func (l *Link) TargetFilePath(workDir ...string) string {
+	targetFilePath := filepath.FromSlash(strings.TrimSuffix(l.LinkFilePath, linkExtension))
+	wd := l.WorkDir
+	if len(workDir) > 0 {
+		wd = workDir[0]
+	}
+	return filepath.Join(wd, targetFilePath)
+}
+
+// IncludeLinkedFiles function includes linked files from the source
+// directory to the target directory.
+// It returns a slice of Link structs representing the included files.
+// It also updates the checksum of the linked files.
+// Both directories must be relative to the root.
+func IncludeLinkedFiles(fromDir, toDir string) ([]Link, error) {
+	links, err := ListLinkedFiles(fromDir)
+	if err != nil {
+		return nil, fmt.Errorf("including linked files failed: %w", err)
+	}
+	for _, l := range links {
+		if _, err := l.UpdateChecksum(); err != nil {
+			return nil, fmt.Errorf("could not update checksum for file %v: %w", l.LinkFilePath, err)
+		}
+		targetFilePath := l.TargetFilePath(toDir)
+		if err := CopyFile(
+			filepath.Join(l.WorkDir, filepath.FromSlash(l.IncludedFilePath)),
+			targetFilePath,
+		); err != nil {
+			return nil, fmt.Errorf("could not write file %v: %w", targetFilePath, err)
+		}
+	}
+
+	return links, nil
+}
+
+// ListLinkedFiles function returns a slice of Link structs representing linked files.
+func ListLinkedFiles(fromDir string) ([]Link, error) {
+	var linkFiles []string
+	if err := filepath.Walk(
+		filepath.FromSlash(fromDir),
+		func(path string, info os.FileInfo, err error) error {
+			if err != nil {
+				return err
+			}
+			if !info.IsDir() && strings.HasSuffix(info.Name(), linkExtension) {
+				linkFiles = append(linkFiles, path)
+			}
+			return nil
+		}); err != nil {
+		return nil, err
+	}
+
+	links := make([]Link, len(linkFiles))
+
+	for i, f := range linkFiles {
+		l, err := NewLinkedFile(filepath.FromSlash(f))
+		if err != nil {
+			return nil, fmt.Errorf("could not initialize linked file %v: %w", f, err)
+		}
+		links[i] = l
+	}
+
+	return links, nil
+}
+
+// CopyFile function copies a file from to to inside the root.
+func CopyFile(from, to string) error {
+	from = filepath.FromSlash(from)
+	source, err := os.Open(from)
+	if err != nil {
+		return err
+	}
+	defer source.Close()
+
+	to = filepath.FromSlash(to)
+	dir := filepath.Dir(to)
+	if _, err := os.Stat(dir); os.IsNotExist(err) {
+		if err := os.MkdirAll(dir, 0700); err != nil {
+			return err
+		}
+	}
+	destination, err := os.Create(to)
+	if err != nil {
+		return err
+	}
+	defer destination.Close()
+
+	_, err = io.Copy(destination, source)
+	return err
+}
+
+// WriteFile function writes a byte slice to a file inside the root.
+func WriteFile(to string, b []byte) error {
+	to = filepath.FromSlash(to)
+	if _, err := os.Stat(filepath.Dir(to)); os.IsNotExist(err) {
+		if err := os.MkdirAll(filepath.Dir(to), 0700); err != nil {
```

Comment:
We create files with `0644`, shouldn't we also create directories with `0755`?
```suggestion
		if err := os.MkdirAll(filepath.Dir(to), 0755); err != nil {
```